lib/models/User.js

```diff
@@ -8,18 +8,23 @@ var cypher = new Mcrypt('rijndael-128', 'ecb');
 var key = require('./../utils/get-mcrypt-key.js')();
 cypher.open(key);
 
+function decryptUsername(username) {
+    return cypher.decrypt(new Buffer(username.trim(), 'base64'))
+        .toString().replace(/\0/g, '');
+}
+
 module.exports = function(bookshelf) {
     return bookshelf.extend({
         tableName: 'cas_users',
+        idAttribute: 'username',
         initialize: function() {
-            this.on('fetched', this.decryptAttributes);
+            this.on('saved', this._utils.saveHist, this);
```

Comment:
Any models which specify a custom `initialize` fn will need to explicitly include the `saveHist` event listener.